CRM/Contribute/BAO/Contribution.php

```diff
@@ -5022,9 +5029,11 @@ public static function checkLineItems(&$params) {
       }
 
       if (!CRM_Utils_Money::equals($totalAmount, $lineItemAmount, $currency)) {
-        throw new CRM_Contribute_Exception_CheckLineItemsException();
+        return FALSE;
```

Comment:
@mattwire earlier the intent might be to have native exception handler for line-item errors. But then if I go with the change it is doing the same thing but old code throw native exception instead of API exception although it is used only in order.create api. So I want to know more why do you think we should get rid of CRM_Contribute_Exception_CheckLineItemsException?

Reply:
I think there was an older PR where we discussed this.  I didn't really like the return value of this function being changed but I could see why @omarabuhussein wanted to - it makes the function easier to test.  But if there is a reason why we should keep the exception then it should stay.

Reply:
Can't testing convenience be solved just by handling the exception in the test? Nuanced exceptions is a pattern we have been increasing not decreasing.

Reply:
@eileenmcnaughton but that's does not solve how we can write 'positive' tests for this method : https://github.com/civicrm/civicrm-core/pull/12372#issuecomment-401403151

Reply:
@omarabuhussein well if negative is an exception then positive is no exception

Reply:
yeah I realize that, it is just I don't like testing for "no exception thrown", something feels wrong about it to me. But if everyone insist to keep the exception (as it seem the case here) then I guess I have no other option.